lib/kubeclient/common.rb

```diff
@@ -63,6 +70,9 @@ def initialize_client(
       @ssl_options = ssl_options
       @auth_options = auth_options
       @socket_options = socket_options
+      # Allow passing partial timeouts hash, without unspecified
+      # @timeouts[:foo] == nil resulting in infinite timeout.
```

Comment:
:+1: for code.
not sure we should merge all of the other defaults - lets discuss there

why would `@timeouts[:foo] == nil` result in infinite timeout?